src/Controller/AuthController.php

```diff
@@ -63,17 +69,76 @@ public function register(Request $request, EntityManagerInterface $manager, User
 
     /**
      * @Route("/forgot-password", name="forgot_password")
+     * @Security("is_granted('IS_ANONYMOUS')")
      */
-    public function forgotPassword()
+    public function forgotPassword(Request $request, \Swift_Mailer $mailer, UserRepository $repository, TokenGeneratorInterface $tokenGenerator, EntityManagerInterface $manager)
     {
-        return $this->render('auth/forgot-password.html.twig');
+        $form = $this->createForm(ForgotPasswordType::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $user = $repository->findOneBy(['email' => $form->getData()->getEmail()]);
+
+            if ($user) {
+                $token = $tokenGenerator->generateToken();
+                $user->setResetToken($token);
+                $manager->flush();
+                $url = $this->generateUrl('reset_password', ['token' => $token], UrlGeneratorInterface::ABSOLUTE_URL);
+
+                $message = $mailer->createMessage()
+                    ->setSubject('Oubli de mot de passe - Réinisialisation')
+                    ->setFrom('help@snowtricks.fr')
+                    ->setTo($user->getEmail())
+                    ->setBody(
+                        $this->renderView(
+                            'auth/emails/resetPasswordEmail.html.twig',
+                            [
+                            'user'=>$user,
+                            'url'=>$url,
+                            'token' => $token
+                        ]
+                        ),
+                        'text/html'
+                    );
+
+                $mailer->send($message);
+            }
+
+            $this->addFlash('success', 'Tu recevras un email si il existe dans la base de données !');
+
+            return $this->redirectToRoute('login');
+        }
+
+        return $this->render('auth/forgot-password.html.twig', [
+            'form' => $form->createView()
+        ]);
     }
 
     /**
-     * @Route("/reset-password", name="reset_password")
+     * @Route("/reset-password/{token}", name="reset_password")
+     * @Security("is_granted('IS_ANONYMOUS')")
      */
-    public function resetPassword()
+    public function resetPassword(Request $request, UserRepository $repository, EntityManagerInterface $manager, string $token, UserPasswordEncoderInterface $encoder)
     {
-        return $this->render('auth/reset-password.html.twig');
+        $user = $repository->findOneBy(['resetToken' => $token]);
+        $form = $this->createForm(ResetPasswordType::class, $user);
+        $form->handleRequest($request);
+
+        if ($user && $form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $hash = $encoder->encodePassword($user, $data->getPassword());
+            $user->setPassword($hash);
+            $user->setResetToken(null);
+            $manager->persist($user);
```

Comment:
no need to persist, the user is already in the database

Reply:
understood